components/common/src/ui.rs

```diff
@@ -43,6 +44,169 @@ pub const NONINTERACTIVE_ENVVAR: &str = "HAB_NONINTERACTIVE";
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum UISymbol {
+    UpArrow,
+    FingerPoint,
+    CheckMark,
+    BoxedCheckMark,
+    Omega,
+    BoxedX,
+    RightArrow,
+    Cloud,
+    DownArrow,
+    Elipses,
+    DottedTriangle,
+    RightShift,
+    Star,
+    SlashedZero,
+    ErrorX,
+}
+
+impl UISymbol {
+    pub fn to_str(&self) -> &str {
+        let style = if let Ok(s) = env::var(SYMBOL_STYLE_ENVVAR) {
+            match UISymbolStyle::from_str(&s) {
+                Ok(style) => style,
+                Err(e) => {
+                    eprintln!("{}\nSetting UISymbolStyle to {}",
+                              e,
+                              UISymbolStyle::default());
+                    UISymbolStyle::default()
+                }
+            }
+        } else {
+            UISymbolStyle::default()
+        };
+
+        match style {
+            UISymbolStyle::Ascii => {
+                match *self {
+                    UISymbol::UpArrow => "/^\\",
+                    UISymbol::FingerPoint => "-->",
+                    UISymbol::CheckMark => "[x]",
+                    UISymbol::BoxedCheckMark => "[#]",
+                    UISymbol::Omega => "-->",
+                    UISymbol::BoxedX => "-X-",
+                    UISymbol::RightArrow => "-->",
+                    UISymbol::Cloud => "-->",
+                    UISymbol::DownArrow => "  >",
+                    UISymbol::Elipses => "...",
+                    UISymbol::DottedTriangle => "???",
+                    UISymbol::RightShift => " >>",
+                    UISymbol::Star => "***",
+                    UISymbol::SlashedZero => "  0",
+                    UISymbol::ErrorX => "XXX",
+                }
+            }
+            UISymbolStyle::Limited => {
+                match *self {
+                    UISymbol::UpArrow => "↑",
+                    UISymbol::FingerPoint => "→",
+                    UISymbol::CheckMark => "√",
+                    UISymbol::BoxedCheckMark => "⌂",
+                    UISymbol::Omega => "Ω",
+                    UISymbol::BoxedX => "░",
+                    UISymbol::RightArrow => "→",
+                    UISymbol::Cloud => "⌂",
+                    UISymbol::DownArrow => "↓",
+                    UISymbol::Elipses => "…",
+                    UISymbol::DottedTriangle => "‼",
+                    UISymbol::RightShift => "»",
+                    UISymbol::Star => "≡",
+                    UISymbol::SlashedZero => "Ø",
+                    UISymbol::ErrorX => "XXX",
+                }
+            }
+            UISymbolStyle::Full => {
+                match *self {
+                    UISymbol::UpArrow => "↑",
+                    UISymbol::FingerPoint => "☛",
+                    UISymbol::CheckMark => "√",
+                    UISymbol::BoxedCheckMark => "☑",
+                    UISymbol::Omega => "Ω",
+                    UISymbol::BoxedX => "☒",
+                    UISymbol::RightArrow => "→",
```

Comment:
This is used somewhat inconsistently for all the following statuses
```
Demoting
Found
Generated
Promoting
Using
Writing
```
I think it would be more useful UX to make the "-ing" statuses use the same glyph we use elsewhere for in-progress actions and the complete actions can use `::CheckMark` (or `::Completed`). Also, "Demoted" could use something like

↩︎
LEFTWARDS ARROW WITH HOOK
Unicode: U+21A9 U+FE0E, UTF-8: E2 86 A9 EF B8 8E

or

⎌
UNDO SYMBOL
Unicode: U+238C, UTF-8: E2 8E 8C